frontend/lib/verification/verification_workflow.dart

```diff
@@ -0,0 +1,125 @@
+    _qrScannerContext = qrScannerContext;
+    _openWaitingDialog();
+
+    final client = GraphQLProvider.of(_qrScannerContext).value;
+    try {
+      final card = processQrCodeContent(rawQrContent);
+      final valid = await queryServerVerification(client, card);
+      if (!valid) {
+        await _onError("Die zu prüfende Karte konnte vom Server nicht "
+            "verifiziert werden!", "cardRejected");
+      } else {
+        await _onSuccess(card.cardDetails);
+      }
+    } on ServerVerificationException catch (e) {
+      await _onError("Die eingescannte Ehrenamtskarte konnte nicht verifiziert "
+        "werden, da die Kommunikation mit dem Server fehlschlug.",
+        "verifyRequestError", e);
+    } on QrCodeFieldMissingException catch (e) {
+      await _onError("Die eingescannte Ehrenamtskarte ist nicht gültig, "
+          "da erforderliche Daten fehlen.", "${e.missingFieldName}Missing", e);
+    } on CardExpiredException catch (e) {
+      final dateFormat = DateFormat("dd.MM.yyyy");
+      await _onError("Die eingescannte Karte ist bereits am "
+        "${dateFormat.format(e.expiry)} abgelaufen.", "cardExpired", e);
+    } on QrCodeParseException catch (e) {
+      await _onError("Der Inhalt des eingescannten Codes kann nicht verstanden "
+        "werden. Vermutlich handelt es sich um einen QR-Code, der nicht für "
+        "die Ehrenamtskarte-App generiert wurde.", "invalidFormat", e);
+    } on Exception catch (e) {
+      await _onError("Ein unbekannter Fehler beim Einlesen des QR-Codes ist "
+        "aufgetreten.", "unknownError", e);
+    } finally {
+      // Should already be closed in any case, but we want to really be sure the
+      // dialog eventually is closed.
+      _closeWaitingDialog();
+    }
+  }
+
+  Future<void> _onError(String message,
+      [String errorCode, Exception exception]) async {
+    if (exception != null) {
+      print("Verification failed: ${exception.toString()}");
+    }
+    if (_userCancelled) return;
+    _closeWaitingDialog();
+    await VerificationResultDialog.showFailure(_qrScannerContext, message,
+        errorCode);
+  }
+
+  Future<void> _onSuccess(BaseCardDetails cardDetails) async {
+    if (_userCancelled) return;
+    _closeWaitingDialog();
+    await VerificationResultDialog.showSuccess(_qrScannerContext, cardDetails);
+  }
+
+  void _openWaitingDialog() {
+    showDialog(context: _qrScannerContext, builder: (context) {
+      _waitingDialogContext = context;
+      return AlertDialog(
+        title: Text("Einen Moment bitte …"),
+        content: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Text("Wir überprüfen den gescannten Code für Sie."),
+            SizedBox(height: 12),
+            CircularProgressIndicator()
+        ]),
+        actions: [
+          TextButton(
+              onPressed: () => Navigator.pop(context),
+              child: Text("Abbrechen"))
+        ],
+      );
```

Comment:
Too much text for something that is usually displayed <1 second.
Maybe only show the ProgressIndicator and maybe the cancel button in a custom Dialog with opaque background. (or similar :) )

Reply:
True. I simplified it.